### ISSUE TITLE
Container got stuck after 'docker start xxx' 

### ISSUE THREAD

**Issue by sh00tg0a1** [docker/for-mac#2906]
Diagnostic ID: 1D0BB2CB-F933-4EA8-8112-88F5B2670D78

  - [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have submitted Diagnostics
  - [x] I have included the Diagnostics ID in this Issue
  - macOS Version: macOS: version 10.13.4 (build: 17E199)

### Expected behavior
   Container started

### Actual behavior
   Starting process got stuck, but I can do 'docker ps' or 'docker images' in other shell windows.

### Information

```
Docker for Mac: version: 18.03.1-ce-mac65 (dd2831d4b7421cf559a0881cc7a5fdebeb8c2b98)
macOS: version 10.13.4 (build: 17E199)
logs: /tmp/1D0BB2CB-F933-4EA8-8112-88F5B2670D78/20180516-112045.tar.gz
[OK]     vpnkit
[OK]     vmnetd
[OK]     dns
[OK]     driver.amd64-linux
[OK]     virtualization VT-X
[OK]     app
[OK]     moby
[OK]     system
[OK]     moby-syslog
[OK]     kubernetes
[OK]     files
[OK]     env
[OK]     virtualization kern.hv_support
[OK]     osxfs
[OK]     moby-console
[OK]     logs
[OK]     docker-cli
[OK]     disk

```

### Steps to reproduce the behavior

  1. docker start {some_container}
  2. stuck...

**Comment by akimd** [comment]
Hi,
I don't have your logs.  Are you sure they were uploaded properly?

**Comment by YRM64** [comment]
Tagged #10589, "Sometimes a Docker container gets stuck so that it can't be stopped or removed."
However, there were several use cases with the same problem, and yet, there was no expected behavior or actual behavior noted in #10589, so I can't speak to similarity/dissimilarity in performance output, or behavior. Notwithstanding,  there isn't any output of Docker version in #2906 to do a comparative analysis with #10589.

In #10589, I thought it noteworthy to share that vankhoa011 fixed his problem with a Docker update of 1.5.  Jinyk had a similar issue, and fixed it with increasing memory of the VM from 2GB to 3 GB.  Jayfk commented that rebooting the machine for his images (icings/icinga2) solved his issue.  elhu added that once the server's kernel was upgraded to 3.13.0-73.116, that was a fix.  mulyoved commented that the problem doesn't exist with other identical loads on older Linux machines (3.13.0-24-generic).  ptepper added that he saw the same issues on Ubuntu 14.04.3; tried an upgrade on Ubuntu and wily; experienced negative results, but when he downgraded to 3.13.xx, that fixed the problem.

There's been several reports of this same issue resurfacing when upgrading to Docker 1.9 (via toolbox) on OSX - when the process tries to exit, it gets stuck.

Similar to the current case, bitliner reported that he had a container stick during execution, and now he can't remove it, or stop it.  There are many similarities/dissimilarities between the two cases, far too many to mention. There are also references drawn to other use cases similar to both, such as #12738, Docker #18180, Docker #19552, Docker #18180, Docker #22357, Docker #22312, Docker/Swarm #2304, and docker #26017.  And, as @thaJeztah indicated in #10589, he was closing because of a collection of unrelated problems, reiterating that #22423 matches most of the descriptions complained about.

Based upon the factual use cases, and available information, I draw the foregoing conclusion:
Sh00tg0a1 indicated that a "docker ps" could be done, but there was no mention as to whether or not a re-start policy had been put in place, or implemented.  And, the reason I think that's significant is because I'm not convinced that shOOtgOa1's container actually stopped running.  ShOOtgOa1's own words were, actual behavior: "starting process got stuck."  That implies a 'start' (some action).  To meet the qualifying criteria to use a re-start policy when using the docker run command, one must have at least started the container.  The implication here in this case is that the container was started, and then got stuck in the start process.  A re-start policy only takes effect after a successful re-start.  My contribution is that shOOtgOa1 could probably have benefited by creating a restart policy for the container(s), utilizing a --restart flag when using the docker run command.  As long as the container has initially started, commands to be utilized include:  restart on failure (which manifests as a non-zero code), always restart the container if it stops, and the ones that I especially cater to include: Docker inspect <task or container>, Docker container rm infinite (if one is forced to do so, due to long periods of inactivity), Docker container start nginx, Docker container restart nginx, Docker container unpause nginx, connecting a running container to a network (in case it's a network problem), and if you need to remove the container and it's volume altogether, Docker container rm nginx.

**Comment by akimd** [comment]
No diagnostics.
